### PR TITLE
Add NP survey

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205301853498999/f

### Description
Add support for the NP survey. See asana task for more info

### Steps to test this PR
Follow the test steps in [here](https://app.asana.com/0/0/1205318335997484/f).

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3479** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3479" target="_blank">Graphite</a> 
    * **PR #3480** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3480" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3480?utm_source=stack-comment).

File: app/src/main/java/com/duckduckgo/app/survey/api/SurveyService.kt [hunk 1]
```diff
@@ -27,6 +27,6 @@ interface SurveyService {
     @GET("https://staticcdn.duckduckgo.com/survey/v2/survey-mobile.json")
     fun survey(): Call<SurveyGroup?>
 
-    @GET("https://staticcdn.duckduckgo.com/survey/apptp/retention-survey-mobile.json")
-    fun surveyAppTp(): Call<SurveyGroup?>
+    @GET("https://staticcdn.duckduckgo.com/survey/netp/waitlist-survey-mobile.json")
+    fun surveyNetPWaitlistBeta(): Call<SurveyGroup?>
```
Comment (nalcalag) [inline app/src/main/java/com/duckduckgo/app/survey/api/SurveyService.kt:31]: Will the appTP survey no longer be needed? Or is the logic already included in the `survey()` call to handle it?

**Review by karlenDimla — APPROVED** [review]
The client code part works as expected. However, the survey was a bit flaky. Sometimes it just shows a blue screen after a few answers.

File: app/src/main/java/com/duckduckgo/app/survey/api/SurveyService.kt (hunk 1 above)
Comment (aitorvs) [reply app/src/main/java/com/duckduckgo/app/survey/api/SurveyService.kt:31]: yeah, AppTP survey no longer used

**Pull request merged**